test/jdk/java/net/httpclient/http2/server/Http2TestServerConnection.java

```diff
@@ -945,6 +946,12 @@ private void handlePush(OutgoingPushPromise op) throws IOException {
         nextPushStreamId += 2;
         pp.streamid(op.parentStream);
         writeFrame(pp);
+        if (pp.getFlags() != HeadersFrame.END_HEADERS && op.hasContinuations()) {
```

Comment:
Maybe you could just drop `op.hasContinuations()`

Reply:
I had it like this to account for a case where a test could incorrectly add continuations to an OutgoingPushPromise without correctly setting the END_HEADERS flag on the PushPromiseFrame. This could possibly be handled in OutgoingPushPromise though.